### PR TITLE
fix __torch_function__ bug in getindex that causes an error not set exception

### PR DESCRIPTION
**Pull request by zdevito**
Without this change, the test will fail with `error return without exception set`. This is because of a difference between what `wrapTuple` considers a tuple (among other things, lists of Tensors under size 32!) and what handle_torch_function_indexing considers a tuple (just tuples).

```
SystemError                               Traceback (most recent call last)
[<ipython-input-4-d346ff1cd397>](https://op8nhaudj1-496ff2e9c6d22116-0-colab.googleusercontent.com/outputframe.html?vrz=colab-20220601-060046-RC00_452259785#) in <module>()
      7 a = torch.rand(3)
      8 
----> 9 a[[T(), T()]]

SystemError: error return without exception set
```

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #78781

### CODE REVIEW

File: torch/csrc/autograd/python_variable_indexing.cpp [hunk 1]
```diff
@@ -318,7 +318,7 @@ PyObject* THPVariable_getitem(PyObject* self, PyObject* index) {
   variable_list variableIndices;
   int64_t specified_dims = count_specified_dimensions(holder.get());
   if (specified_dims == -1) {
-    return handle_torch_function_indexing(self, index);
+    return handle_torch_function_indexing(self, holder.get());
```
Comment (albanD) [inline torch/csrc/autograd/python_variable_indexing.cpp:321]: Wasn't the point here that the user's `__torch_function__` would be called with the original argument and not the newly created tuple?

Also can you give more details how this was causing an error?

Comment (albanD) [reply torch/csrc/autograd/python_variable_indexing.cpp:321]: So that function unpacks tuple but not list ? (I tried to look but it is a bit hard to follow)

Comment (ezyang) [reply torch/csrc/autograd/python_variable_indexing.cpp:321]: Normalization is OK, we do this occasionally, there is not an invariant that the arguments match up exactly. On the buggy case this logic

```
  if (PyTuple_Check(index)) {
    index_tup = py::reinterpret_borrow<py::object>(index);
  }
  else {
    index_tup = py::make_tuple(py::handle(index));
  }
```

causes index_tup to be `([T(), T()],)` but only the first tuple is iterated over. We could fix this but I'm happy to take this PR as is.

**Review by ezyang — APPROVED** [review]
